components/track/TrackChangeLocale.vue

```diff
@@ -0,0 +1,123 @@
+<script setup lang="ts">
+import { LanguageEnum } from "@bcc-code/bmm-sdk-fetch";
+import { Menu, MenuButton, MenuItem, MenuItems } from "@headlessui/vue";
+
+const expanded = ref(false);
+const { currentTrack, replaceCurrent } = useNuxtApp().$mediaPlayer;
+const { t } = useI18n();
+
+const changeLanguage = async (lang: LanguageEnum) => {
+  expanded.value = false;
+  if (!currentTrack.value) return;
+
+  try {
+    const reloadedTrack = await useTrackIDWithLanguage(lang, {
+      id: currentTrack.value.id,
+    });
+    if (!reloadedTrack.data.value) return;
+    replaceCurrent(reloadedTrack.data.value);
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const trackLanguages = currentTrack?.value?.languages || [];
+
+const getUserLanguages = () => {
+  const contentLangs = contentLanguageStore()
+    .contentLanguages as LanguageEnum[];
+  return contentLangs.filter((lang) => trackLanguages.includes(lang));
+};
+
+const getAvailableLanguages = () => {
```

Comment:
It seems only the length of this is necessary. Could this then be solved differently?

Reply:
yeah, it is actually trackLanguages.length... I removed this function